src/vivarium/framework/engine.py

```diff
@@ -274,8 +270,22 @@ def report(self, print_results: bool = True) -> Dict[str, Any]:
                 float_format=lambda x: f"{x:.2f}",
             )
             self._logger.info("\n" + performance_metrics)
+        for measure, df in metrics.items():
```

Comment:
What if each observation was also registered with a `report` method that was called on the `simulation_end` event? For all observers that we have so far it would just be what you have here, but it would enable additional flexibility if we had an observer that wanted to write results in a different way (e.g. to a parquet or hdf file).

Reply:
Hmm, that's a good idea - I'll implement and see how it goes

Reply:
As discussed, this will happen in a follow-up PR